### DISCUSSION TITLE
(Battery) Power supply

### DISCUSSION THREAD

**Discussion by thebino** [biologist79/ESPuino#55] [General]
Hi,
what are your experience with battery power supply? I would like to integrate a power supply with LiPo to make a portable version. I found this circuit Diagram but I'm not sure, if I can wire the "Power IN" directly instead of the GPIO of the ESP32.
![ESP32-Power-Supply-Circuit](https://user-images.githubusercontent.com/841681/104417171-414c8300-5575-11eb-9a01-1b0f68c3dce4.jpeg)

**Comment by biologist79** [comment]
Hi,

yes and it's covered by my documentation. 
Short answer is: use Wemos Lolin32 / Lolin D32 / Lolin D32 pro as it's already integrated there.

**Reply by thebino** [reply]
But without replacing the ESP? The ESP is already soldered on the PCB and it would be a mess to unsolder and replace it.

**Comment by biologist79** [comment]
I don't get your point. It's all onboard: https://www.wemos.cc/en/latest/d32/index.html
LiPo is connected via JST PH2.0 connector.

**Comment by thebino** [comment]
The ESP32 I used is soldered on the Tonuino PCB already and I would like to avoid replacing it which would need to unsolder 38 pins.
![PXL_20210113_131600145](https://user-images.githubusercontent.com/841681/104458813-22b5ae80-55ac-11eb-91cb-a34fc4082303.jpg)

**Comment by biologist79** [comment]
a) I'd always recommend to solder a female pin-socket that is used to plug-in the ESP32-develboard.
b) I wouldn't use this ESP32-develboard with battery-mode because, even in deepsleep, it's quiescent current is way too high (>10 mA or so). So, as already outlined, I'd recommend a Lolin-develboard.

**Reply by thebino** [reply]
Thanks a lot.

Also good point to use a socket.

**Comment by biologist79** [comment]
I still have two left-overs from the first revision of this PCB: https://github.com/biologist79/Tonuino-ESP32-I2S/tree/master/PCBs/Wemos%20Lolin32
By mistake I switched two labels for RFID and printed 4.7kOhms on the PCB whereas only 1k is needed. But beside of that it's fine. But, unless to don't want to solder wires directly onto the PCB you need JST PH connectors (2mm-pitch). Soldered it looks like this: https://github.com/biologist79/Tonuino-ESP32-I2S/blob/master/PCBs/Wemos%20Lolin32/Pictures/Soldered2_rev1.jpg

Write me an eMail in case you're interested in such a PCB.

**Comment by biologist79** [comment]
Board is this one: https://de.aliexpress.com/item/4000007523759.htm or https://www.ebay.de/itm/WEMOS-LOLIN32-V1-0-0-ESP32-Rev1-ESP-WROOM-32-WIFI-Bluetooth-Board/352994042051
Alternatively use Lolin D32, but seems to be unavailable @Aliexpress and is really expensive in Germany.
Didn't have time yet to design a PCB for Lolin D32 pro: https://de.aliexpress.com/item/32883116057.html
 => Pro: has PSRAM (that's good for webstreams)  Con: lacks GPIO 16+17

**Comment by biologist79** [comment]
Make sure to not order Lolin32 lite. Have been told it provides a weak LDO (don't know if that's true).
